src/core/config/npo/chunk.rs

```diff
@@ -45,6 +45,14 @@ impl<A> Chunk<A> {
         vec
     }
 
+    pub fn map<B>(&self, f: &mut impl FnMut(&A) -> B) -> Chunk<B> {
+        match self {
+            Chunk::Empty => Chunk::Empty,
+            Chunk::Append(a, rc) => Chunk::Append(f(a), Rc::new(rc.map(f))),
+            Chunk::Concat(a, b) => Chunk::Concat(Rc::new(a.map(f)), Rc::new(b.map(f))),
+        }
+    }
```

Comment:
The `map` operator isn't lazy like the rest of the operators. For very large chunks this will allocate a lot of memory.

Reply:
The current usage is to generate the result of npo and we need this output anyway and laziness is not doing much there. 

But I'm open to the suggestions if we can implement this in a lazy way. I see the issues with Rc wrappers and reference/owned value.